src/server/system_services/bucket_server.js

```diff
@@ -1142,20 +1142,32 @@ function get_cloud_buckets(req) {
                 return storage.getBuckets()
                     .then(data => data[0].map(bucket =>
                         _inject_usage_to_cloud_bucket(bucket.name, connection.endpoint, used_cloud_buckets)));
-            } else { //else if AWS
+            } else {
+                var access_key;
+                var secret_key;
+                if (connection.aws_sts_arn) {
+                    const creds = await cloud_utils.generate_aws_sts_creds(connection, "get_cloud_buckets_session");
+                    access_key = creds.accessKeyId;
+                    secret_key = creds.secretAccessKey;
+                    connection.sessionToken = creds.sessionToken;
+                } else {
+                    access_key = connection.access_key.unwrap();
+                    secret_key = connection.secret_key.unwrap();
+                }
                 var s3 = new AWS.S3({
                     endpoint: connection.endpoint,
-                    accessKeyId: connection.access_key.unwrap(),
-                    secretAccessKey: connection.secret_key.unwrap(),
+                    accessKeyId: access_key,
+                    secretAccessKey: secret_key,
+                    sessionToken: connection.sessionToken,
                     signatureVersion: cloud_utils.get_s3_endpoint_signature_ver(connection.endpoint, connection.auth_method),
                     s3DisableBodySigning: cloud_utils.disable_s3_compatible_bodysigning(connection.endpoint),
                     httpOptions: {
                         agent: http_utils.get_unsecured_agent(connection.endpoint)
                     }
                 });
-                const used_cloud_buckets = cloud_utils.get_used_cloud_targets(['AWS', 'S3_COMPATIBLE', 'FLASHBLADE', 'IBM_COS'],
+               const used_cloud_buckets = cloud_utils.get_used_cloud_targets(['AWS', 'S3_COMPATIBLE', 'FLASHBLADE', 'IBM_COS'],
```

Comment:
also, AWS_STS is not missing here?

Reply:
I think its not required, since it is of type AWS only.

Reply:
Added AWS_STS in the get_used_cloud_targets as list